contracts/andromeda_gumball/src/contract.rs

```diff
@@ -0,0 +1,881 @@
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    let contract = ADOContract::default();
+    match msg {
+        ExecuteMsg::AndrReceive(msg) => contract.execute(deps, env, info, msg, execute),
+        ExecuteMsg::Mint(mint_msg) => execute_mint(deps, env, info, mint_msg),
+        ExecuteMsg::Buy {} => execute_buy(deps, env, info),
+        ExecuteMsg::SaleDetails {
+            price,
+            max_amount_per_wallet,
+            recipient,
+        } => execute_sale_details(deps, env, info, price, max_amount_per_wallet, recipient),
+        ExecuteMsg::SwitchStatus {} => execute_switch_status(deps, info),
+        // ExecuteMsg::SetContractAddress {
+        //     andromeda_cw721_contract,
+        // } => execute_switch_contract_address(deps, info, andromeda_cw721_contract),
+    }
+}
+// fn execute_switch_contract_address(
+//     deps: DepsMut,
+//     info: MessageInfo,
+//     msg: AndrAddress,
+// ) -> Result<Response, ContractError> {
+//     let contract = ADOContract::default();
+
+//     require(
+//         contract.is_contract_owner(deps.storage, info.sender.as_str())?,
+//         ContractError::Unauthorized {},
+//     )?;
+//     CW721_CONTRACT.save(deps.storage, &msg)?;
+//     Ok(Response::new().add_attribute("action", "set cw721 address"))
+// }
+fn execute_switch_status(deps: DepsMut, info: MessageInfo) -> Result<Response, ContractError> {
+    let contract = ADOContract::default();
+    let mut status = STATUS.load(deps.storage)?;
+    require(
+        contract.is_contract_owner(deps.storage, info.sender.as_str())?,
+        ContractError::Unauthorized {},
+    )?;
+    // in case owner forgot to set the state, can't allow purchasing without the sale details set
+    let state = STATE.may_load(deps.storage)?;
+    require(state.is_some(), ContractError::PriceNotSet {})?;
+    // Automatically switch to opposite status
+    if status {
+        status = false;
+    } else {
+        status = true;
+    }
+    STATUS.save(deps.storage, &status)?;
+    Ok(Response::new().add_attribute("action", "Switched Status"))
+}
+fn execute_sale_details(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    price: Coin,
+    max_amount_per_wallet: Option<Uint128>,
+    recipient: Recipient,
+) -> Result<Response, ContractError> {
+    let contract = ADOContract::default();
+    let status = STATUS.load(deps.storage)?;
+    // Check status, can't change sale details while buying is allowed
+    require(!status, ContractError::Refilling {})?;
+    // Check authority
+    require(
+        contract.is_contract_owner(deps.storage, info.sender.as_str())?,
+        ContractError::Unauthorized {},
+    )?;
+    // Check valid amount
+    require(
+        price.amount > Uint128::from(0_u64),
+        ContractError::InvalidZeroAmount {},
+    )?;
+    // Check valid denomination
+    require(
+        price.denom == *"uusd",
```

Comment:
nit: Move `"uusd"` to a constant. Do we want to allow the user to specify which native fund is being used for the sale?

Reply:
I think defining the denom is a good approach